src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs

```diff
@@ -234,6 +234,12 @@ private Task UpdateReferencesAync(Project project, CancellationToken cancellatio
             private async Task UpdateReferenceAsync(
                 Project project, PortableExecutableReference reference, CancellationToken cancellationToken)
             {
+                if (cancellationToken.IsCancellationRequested)
```

Comment:
A project can have many, perhaps 100 references. Rather than throw one cancellation exception per *reference*, we now throw one per *project*.